File: implementations/java/array/MyArray.java
public class MyArray {
    private int[] data;
    private int size; // 실제로 채워진 원소 개수 (data.length는 "용량")

    public MyArray() {
        data = new int[10];
        size = 0;
    }

    public int get(int i) {
        // TODO: i번째 원소를 반환하기 (O(1))
        return data[i];
    }

    public void set(int i, int value) {
        // TODO: i번째 자리를 value로 덮어쓰기 (O(1))
        data[i] = value;
    }

    public void insert(int i, int value) {
        if (size == data.length) {
            grow();
        }
        for (int r = size; r > i; r--) {
            data[r] = data[r - 1];
        }
        data[i] = value;
        size++;
    }

    public void delete(int i) {
        // TODO: i번째를 지우고 뒤 원소들을 한 칸씩 당기기 (O(n))
        // insert와 반대 방향 - 여기서는 앞에서 뒤로 당기면 됨
        // [1,2,3,4,5], size=5 에서 인덱스 2를 지우면 [1,2,4,5], size=4
        for (int r = i; r < size - 1; r++) {
            data[r] = data[r+1];
        }
        size--;
    }

    private void grow() {
        int[] bigger = new int[data.length * 2];
        for (int r = 0; r < data.length; r++) {
            bigger[r] = data[r];
        }
        data = bigger;
    }

    public static void main(String[] args) {
        MyArray arr = new MyArray();
        for (int v : new int[] {1, 2, 3, 4, 5}) {
            arr.insert(arr.size, v);
        }
        arr.insert(2, 11);
        for (int r = 0; r < arr.size; r++) {
            System.out.print(arr.get(r) + " ");
        }
        System.out.println();
    }
}

File: implementations/python/array/my_array.py
class MyArray:
    def __init__(self):
        self._data = []

    def get(self, i):
        return self._data[i]

    def set(self, i, value):
        self._data[i] = value

    def insert(self, i, value):
        # TODO: 뒤 원소를 한 칸씩 밀고 i 자리에 value를 넣기
        # [1,2,3,4,5]가 있다고 생각하고
        # 모든 자리를 뒤로 밀고 0에 값 넣기
        self._data.append(0)          # (1,2,3,4,5,0)
        t_len = len(self._data)-2 # 6-2 (0,1,2,3,4,5)
        for r in range(t_len , i-1 , -1):
            self._data[r+1] = self._data[r]
        self._data[i] = value
        print(f"{i}번째 인덱스에 추가됨: {self.get(i)}")
        print(self._data)

    def delete(self, i):
        # TODO: i 자리를 지우고 뒤 원소를 한 칸씩 당기기
        # [1,2,3,4,5]가 있다고 생각하고
        # (0,1,2,3,4)
        # 3을 지워보자
        # [1,2,4,5]
        # (0,1,2,3)
        # 옮기고 맨 뒤에 삭제
        t_len = len(self._data)-1 # 4
        for r in range(i, t_len ): # 2, 4(3)
            self._data[r] = self._data[r+1]
        self._data.pop()
        print(f'삭제된 후 array {self._data}')


arr = MyArray()
for v in [1,2,3,4,5]:
    arr._data.append(v)
arr.insert(2,99)
    